src/functions.rs

```diff
@@ -169,6 +173,71 @@ pub mod stdlib {
     use super::Function;
     use super::Parameters;
 
+    /// The implementation of the standard [`eq`][`crate::reference::functions#eq`] function.
+    pub struct Eq;
+
+    impl Function for Eq {
+        fn call(
+            &self,
+            _graph: &mut Graph,
+            _source: &str,
+            parameters: &mut dyn Parameters,
+        ) -> Result<Value, ExecutionError> {
+            let left = parameters.param()?;
+            let right = parameters.param()?;
+            parameters.finish()?;
+
+            match &left {
+                Value::Null => match right {
+                    Value::Null => return Ok(true.into()),
+                    _ => return Ok(false.into()),
+                },
+                Value::Boolean(left) => match &right {
+                    Value::Null => return Ok(false.into()),
+                    Value::Boolean(right) => return Ok((left == right).into()),
+                    _ => {}
+                },
+                Value::Integer(left) => match &right {
+                    Value::Null => return Ok(false.into()),
+                    Value::Integer(right) => return Ok((left == right).into()),
+                    _ => {}
+                },
+                Value::String(left) => match &right {
+                    Value::Null => return Ok(false.into()),
+                    Value::String(right) => return Ok((left == right).into()),
+                    _ => {}
+                },
+                Value::List(left) => match &right {
+                    Value::Null => return Ok(false.into()),
+                    Value::List(right) => return Ok((left == right).into()),
+                    _ => {}
+                },
+                Value::Set(left) => match &right {
+                    Value::Null => return Ok(false.into()),
+                    Value::Set(right) => return Ok((left == right).into()),
+                    _ => {}
+                },
+                Value::SyntaxNode(left) => match &right {
+                    Value::Null => return Ok(false.into()),
+                    Value::SyntaxNode(right) => return Ok((left == right).into()),
+                    _ => {}
+                },
+                Value::GraphNode(left) => match &right {
+                    Value::Null => return Ok(false.into()),
+                    Value::GraphNode(right) => return Ok((left == right).into()),
+                    _ => {}
+                },
+            };
+            Err(ExecutionError::FunctionFailed(
+                "eq".into(),
+                format!(
+                    "Cannot compare values of different types: {} and {}",
+                    left, right
+                ),
+            ))
```

Comment:
I was about to ask if you could use the derived `std::ops::Eq` impl for this, but now I can see why you didn't!

Reply:
I did at first, and then decided that I didn't want to be responsible for that kind of language :P